src/core.c/allomorphs.pm6

```diff
@@ -1,11 +1,39 @@
+my class Allomorph is Str {
+    multi method Bool(::?CLASS:D:) { self.Numeric.Bool }
+
+    method succ(Allomorph:D:) { self.Numeric.succ }
+    method pred(Allomorph:D:) { self.Numeric.pred }
+
+    multi method Str(Allomorph:D:) {
+        nqp::getattr_s(self,Str,'$!value')
+    }
+
+    multi method WHICH(Allomorph:D:) {
+        nqp::box_s(
+          nqp::join('|',nqp::list_s(
+            self.^name,
+            self.Numeric.WHICH,
+            self.Str.WHICH
+          )),
+          ValueObjAt
+        )
+    }
+
+    multi method raku(Allomorph:D:) {
+        nqp::join("",nqp::list_s(
+          self.^name,'.new(',self.Numeric.raku,', ',self.Str.raku,')'
+        ))
+    }
+}
+
 # the uses of add_I in this class are a trick to make bigints work right
-my class IntStr is Int is Str {
+my class IntStr is Allomorph is Int {
     method new(Int:D $i, Str:D $s) {
         my \SELF = nqp::add_I($i, 0, self);
         nqp::bindattr_s(SELF, Str, '$!value', $s);
-        SELF;
+        SELF
     }
-    multi method ACCEPTS(IntStr:D: Any:D \a) {
+    multi method ACCEPTS(IntStr:D: Any:D \a) is default {
```

Comment:
Not thrilled to see `is default`; what was the conflict?

Reply:
$ r '<42> ~~ <66>'
Ambiguous call to 'ACCEPTS(IntStr: IntStr)'; these signatures all match:
  (Str:D: Str:D \other, *%_)
  (IntStr:D: Any:D \a, *%_)

Reply:
Ah yes, those are. Though we could add an `IntStr:D: Str:D` candidate to overcome that.

Reply:
Especially as we *already* have a branch for `Str` anyway.

Reply:
> Especially as we already have a branch for Str anyway.

I'm not quite sure I follow.  Where would this branch for `Str` be ?

Reply:
Aaah,... I see  :-)

Reply:
Alas:

    # You failed 112 tests of 229
    not ok 107 - .ACCEPTS
    # Failed test '.ACCEPTS'

I think this is caused by the fact that it changes the order in which the tests are performed.  When we have a `Str` candidate, it will be preferred above the `Any` candidate, and thus miss the `Numeric` check.  Will try to see how it works out with a `Numeric` candidate as well.

Reply:
Alas:

    Ambiguous call to 'ACCEPTS(IntStr: IntStr)'; these signatures all match:
      (IntStr:D: Numeric:D \a, *%_)
      (IntStr:D: Str:D \a, *%_)

Reply:
Result of having a Numeric candidate:

    Ambiguous call to 'ACCEPTS(IntStr: Str)'; these signatures all match:
      (Str:D: Str:D \other, *%_)
      (IntStr:D: Any:D \a, *%_)

So, I'm not sure how to fix this other with an `is default`

Reply:
39bc2bdf28 replaces all `ACCEPTS` by a single candidate, but that still needs an `is default` to make tests pass